mage/query-modules/python/igraphalg.md

```diff
@@ -0,0 +1,258 @@
+
+
+| Trait               | Value                                                                                                     |
+| ------------------- | --------------------------------------------------------------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**module**</Highlight>                                                         |
+| **Implementation**  | <Highlight color="#FB6E00">**Python**</Highlight>                                                         |
+| **Graph direction** | <Highlight color="#FB6E00">**directed**</Highlight>/<Highlight color="#FB6E00">**undirected**</Highlight> |
+| **Edge weights**    | <Highlight color="#FB6E00">**weighted**</Highlight>/<Highlight color="#FB6E00">**unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight>                                                     |
+
+## Procedures
+### `get_all_simple_paths(v, to, cutoff)`
+
+Returns all simple paths in the graph `G` from source to target. A simple path is a path with no repeated nodes.
+
+#### Input:
+
+* `v: Vertex` ➡ Starting node for the path.
+* `to: Vertex` ➡ Ending node for the path.
+* `cutoff: int (default=-1)` ➡ Maximum length of path that is considered. If negative, paths of all lengths are considered.
+
+#### Output:
+
+* `path: List[Vertex]` ➡ List of vertices for a certain path. If there are no paths between the source and target within the given cutoff there is no output.
+
+#### Usage:
+```cypher
+MATCH (n:Label), (m:Label)
+CALL igraphalg.get_all_simple_paths(n, m, 5) YIELD *
+RETURN path;
+```
+
+### `spanning_tree(weights, directed)`
+Returns a minimum spanning tree on a graph `G`.
+ A *minimum spanning tree* is a subset of the edges of a connected, graph that connects all of the vertices without any cycles.
+
+#### Input:
+
+* `weights: string (default=NULL)` ➡  Data key to use for edge weights.
+* `directed: bool (default=False)` ➡  Is graph directed or undirected
+
+#### Output:
+
+* `tree: List[List[Vertex]]` ➡  A minimum spanning tree or forest.
+
+#### Usage:
+
+```cypher
+CALL igraphalg.spanning_tree() 
+YIELD *
+RETURN tree;
+```
+
+### `pagerank(damping, weights, directed,implementation)`
+Returns the PageRank of the nodes in the graph.
+
+PageRank computes a ranking of the nodes in graph G based on the structure of the incoming links. It was originally designed as an algorithm to rank web pages.
+
+#### Input:
+
+* `damping: double (default=0.85)` ➡  Damping parameter for PageRank.
+* `weights: string (default="weight")` ➡  Edge data key to use as a weight. If `None`, weights are set to 1.
+* `directed: bool (default=True)` ➡  Is Graph directed or undirected.
+* `implementation: string (default="prpack")` ➡  Algorithm used for calculating PageRank values. The Algorithm can be either `prpack` or `arpack`.
```

Comment:
```suggestion
* `directed: bool (default=True)` ➡  If `true` the graph is directed, otherwise it's undirected
* `implementation: string (default="prpack")` ➡  Algorithm used for calculating PageRank values. The algorithm can be either `prpack` or `arpack`.
```

Reply:
I'm missing what's prpack vs arpack? should this be general knowledge?

Reply:
I am not involved with this PR, but I can give some background info. igraph implements several methods for calculating PageRank scores, and this parameter selects the method. `arpack` phrases the calculation as an eigenvalue problem which is then solved using the ARPACK library. `prpack` uses the PRPACK library, which uses an algebraic method, i.e. phrases as the problem as a set of linear equations.

I believe the reason for having several implementation is historical, as successively better implementations were added. The original method used a plain power iteration, which is not very reliable, so it's now removed. Then came the more robust ARPACK-based implementation, and finally PRPACK, which is usually considerably faster. We kept both the ARPACK and PRPACK-based methods, partly for verification, and partly because there are some edge cases where PRPACK does not work, notably when the damping factor is nearly or exactly 1.0.

Reply:
Thank you so much for explaining!

I'm wondering if we should include a bit more information in the docs as well so people know what to choose, or do we kind of expect them to know? @niko4299 you can decide :)

Reply:
I think we don't need to add this information to docs because people working with PageRank either know about this or when they see these two options will google them and learn more about each method.